apps/sources/models/mixins/document.py

```diff
@@ -0,0 +1,54 @@
+"""Model classes for documents (as sources)."""
+
+from django.db import models
+
+from apps.sources.models.mixins.page_numbers import PageNumbersMixin
+from modularhistory.models import ModelWithComputations, retrieve_or_compute
+from modularhistory.utils.html import soupify
+
+NAME_MAX_LENGTH: int = 100
+LOCATION_INFO_MAX_LENGTH: int = 400
+DESCRIPTIVE_PHRASE_MAX_LENGTH: int = 100
+URL_MAX_LENGTH: int = 200
+
+
+class DocumentMixin(PageNumbersMixin):
+    """A historical document (as a source)."""
+
+    collection = models.ForeignKey(
+        to='sources.Collection',
+        related_name='%(class)s',
+        null=True,
+        blank=True,
+        on_delete=models.SET_NULL,
+    )
+    collection_number = models.PositiveSmallIntegerField(
+        null=True,
+        blank=True,
+        help_text='aka acquisition number',
+    )
+    location_info = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            'Ex: John Alexander Papers, Series 1: Correspondence, 1831-1848, Folder 1'
+        ),
+    )
+    descriptive_phrase = models.CharField(
+        max_length=DESCRIPTIVE_PHRASE_MAX_LENGTH,
+        null=True,
+        blank=True,
+        help_text='e.g., "on such-and-such letterhead" or "signed by so-and-so"',
+    )
+    information_url = models.CharField(
+        max_length=URL_MAX_LENGTH,
+        null=True,
+        blank=True,
+        help_text='URL for information regarding the document',
+    )
+
+    class Meta:
+        """Meta options for the _Engagement model."""
+
+        # https://docs.djangoproject.com/en/3.1/ref/models/options/#model-meta-options.
+        abstract = True
```

Comment:
What is the purpose of this? I see it in [`PageNumberMixin`](https://github.com/ModularHistory/modularhistory/blob/a1bb56fadd7dd46fd139b09e984df0044fd1039c/apps/sources/models/mixins/page_numbers.py#L10) and [`TextualMixin`](https://github.com/ModularHistory/modularhistory/blob/a1bb56fadd7dd46fd139b09e984df0044fd1039c/apps/sources/models/mixins/textual.py#L8), but it doesn't look like either is related to [`_Engagement`](https://github.com/ModularHistory/modularhistory/blob/c4a89b375bc58c36eeefc3fccc2aa8753267ed2a/apps/entities/models/affiliation.py#L12).

Reply:
Oh, or maybe the comment was just copied from somewhere else?

Reply:
Oh, looks like that was copied and pasted. I'll update the docstrings.

`abstract = True` lets Django know that it shouldn't create a table for this model (i.e., it's an abstract model that should only be used for behavior inheritance with the Python objects).